var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -540,9 +540,17 @@ def cmake_args(self):
             projects.append("polly")
             cmake_args.append(define("LINK_POLLY_INTO_TOOLS", True))
 
-        cmake_args.append(from_variant("BUILD_SHARED_LIBS", "shared_libs"))
-        cmake_args.append(from_variant("LLVM_BUILD_LLVM_DYLIB", "llvm_dylib"))
-        cmake_args.append(from_variant("LLVM_USE_SPLIT_DWARF", "split_dwarf"))
+        cmake_args.extend([
+            from_variant("BUILD_SHARED_LIBS", "shared_libs"),
+            from_variant("LLVM_BUILD_LLVM_DYLIB", "llvm_dylib"),
+            from_variant("LLVM_LINK_LLVM_DYLIB", 'llvm_dylib'),
+            from_variant("LLVM_USE_SPLIT_DWARF", "split_dwarf")
+        ])
+
+        # By default on Linux, libc++.so is a linker script, and CMake tries to add the
+        # CMAKE_INSTALL_RPATH to it, which fails, causing installation to fail. The
+        # easiest workaround is to disable it.
+        cmake_args.append(define('LIBCXX_ENABLE_ABI_LINKER_SCRIPT', 'OFF'))
```

Comment:
This one makes me slightly nervous.  The script adds the abi lib when a user links the compiler-rt right?  I'm not saying no, but I want to figure out if this will cause issues with missing symbols for builds using the compiler-rt and the clang abi.

Reply:
The scripts turns `-lc++` into `-lc++ -lc++abi`. With this define turned off, `-lc++` remains `-lc++`. I can check what happens in clang, since the standard flag there is `-stdlib=libstdc++` -- haven't checked what that ends up linking against (hopefully -lc++ -lc++abi then)

Reply:
Hopefully.  Or if libc++ at least has a dt_needed entry for the abi that would be fine too, just want to make sure we don't end up borking it.  Might also want to see what that flag does in a 2-stage build with stage2 having CLANG_USE_LIBCXX and CLANG_USE_LIBCXXABI or whatever they are set, that would be a good stress test for it to build the whole thing against it.